File: brixen-java/src/test/java/org/brixen/test/bean/LoadableBeanTest.java
package org.brixen.test.bean;

import org.brixen.bean.LoadableBean;
import org.brixen.bean.LoadableBeanImpl;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;
import org.openqa.selenium.WebDriver;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;
import static org.testng.Assert.assertNotEquals;

@Test(groups = {"LoadableBeanTestGroup"})
public class LoadableBeanTest {

    private @Mock WebDriver mockDriverOne;
    private @Mock WebDriver mockDriverTwo;

    @BeforeMethod
    public void setUp() {
        MockitoAnnotations.initMocks(this);
    }

    @Test(groups = {"ExpectedExceptionsTestGroup", "BeanExpectedExceptionsTestGroup",
            "LoadableBeanExpectedExceptionsTestGroup"},
            expectedExceptions = { IllegalArgumentException.class },
            expectedExceptionsMessageRegExp = "Cannot specify a load timeout less than 0 seconds"
    )
    public void shouldThrowExceptionForNegativeLoadTimeout() {
        final LoadableBean bean = new LoadableBeanImpl();
        bean.setLoadTimeout(-1);
    }

    @SuppressWarnings("ConstantConditions")
    @Test(groups = {"ExpectedExceptionsTestGroup", "BeanExpectedExceptionsTestGroup",
            "LoadableBeanExpectedExceptionsTestGroup"},
            expectedExceptions = { NullPointerException.class }
    )
    public void shouldThrowExceptionForNullDriver() {
        final LoadableBean bean = new LoadableBeanImpl();
        bean.setDriver(null);
    }

    @Test(groups = {"EqualityNegativeTestGroup", "BeanEqualityNegativeTestGroup",
            "LoadableBeanEqualityNegativeTestGroup"}
    )
    public void shouldNotBeEqual() {
        LoadableBean beanOne = new LoadableBeanImpl();
        LoadableBean beanTwo = new LoadableBeanImpl();

        beanOne.setDriver(mockDriverOne);
        beanTwo.setDriver(mockDriverTwo);

        assertNotEquals(beanOne, beanTwo, "Beans with different driver references should not be equal");

        beanOne = new LoadableBeanImpl();
        beanTwo = new LoadableBeanImpl();

        beanOne.setLoadTimeout(40);

        assertNotEquals(beanOne, beanTwo, "Beans with different load timeouts should not be equal");
    }

    @Test(groups = {"EqualityPositiveTestGroup", "BeanEqualityPositiveTestGroup",
            "LoadableBeanEqualityPositiveTestGroup"}
    )
    public void shouldBeEqual() {
        LoadableBean beanOne = new LoadableBeanImpl();
        LoadableBean beanTwo = new LoadableBeanImpl();

        assertEquals(beanOne, beanTwo, "Two newly constructed beans should be equal before any setters are invoked");

        beanOne.setDriver(mockDriverOne);
        beanTwo.setDriver(mockDriverOne);

        assertEquals(beanOne, beanTwo, "Two beans should be equal if they have the same web driver reference and the " +
                "default load timeout");

        beanOne.setLoadTimeout(40);
        beanTwo.setLoadTimeout(40);

        assertEquals(beanOne, beanTwo, "Two beans should be equal if they have the same web driver reference and the " +
                "same non-default load timeout");

        beanOne = new LoadableBeanImpl();
        beanTwo = new LoadableBeanImpl();

        beanOne.setLoadTimeout(40);
        beanTwo.setLoadTimeout(40);

        assertEquals(beanOne, beanTwo, "Two beans should be equal if they have null web driver reference and the " +
                "same non-default load timeout");
    }
}

File: brixen-dotnet/test/Bean/LoadableBeanTest.cs
using System;
using NUnit.Framework;
using Moq;
using OpenQA.Selenium;
using OpenQA.Selenium.Firefox;
using Org.Brixen.Bean;

namespace Org.Brixen.Tests.Bean {

	[TestFixture]
	[Category("LoadableBeanTestGroup")]
	public class LoadableBeanTest {
		
		[Test]
		[Category("ExpectedExceptionsTestGroup")]
		[Category("BeanExpectedExceptionsTestGroup")]
		[Category("LoadableBeanExpectedExceptionsTestGroup")]
		[ExpectedException("System.ArgumentOutOfRangeException", ExpectedMessage="Cannot specify a load timeout " + 
			"less than 0 seconds", MatchType=MessageMatch.Regex)]
		public void ShouldThrowExceptionForNegativeLoadTimeout() {
			ILoadableBean bean = new LoadableBean();
			bean.LoadTimeout = -1;
		}

		[Test]
		[Category("ExpectedExceptionsTestGroup")]
		[Category("BeanExpectedExceptionsTestGroup")]
		[Category("LoadableBeanExpectedExceptionsTestGroup")]
		[ExpectedException("System.ArgumentNullException", ExpectedMessage="Cannot invoke setter for Driver " + 
			"property with a null parameter", MatchType=MessageMatch.Regex)]
		public void ShouldThrowExceptionForNullDriver() {
			ILoadableBean bean = new LoadableBean();
			bean.Driver = null;
		}

		[Test]
		[Category("EqualityNegativeTestGroup")]
		[Category("BeanEqualityNegativeTestGroup")]
		[Category("LoadableBeanEqualityNegativeTestGroup")]
		public void ShouldNotBeEqual() {
			Mock<IWebDriver> mockDriverOne = new Mock<IWebDriver>();
			Mock<IWebDriver> mockDriverTwo = new Mock<IWebDriver>();

			ILoadableBean beanOne = new LoadableBean();
			ILoadableBean beanTwo = new LoadableBean();

			beanOne.Driver = mockDriverOne.Object;
			beanTwo.Driver = mockDriverTwo.Object;

			Assert.AreNotEqual(beanOne, beanTwo, "Beans with different driver references should not be equal");

			beanOne = new LoadableBean();
			beanTwo = new LoadableBean();

			beanOne.LoadTimeout = 40;

			Assert.AreNotEqual(beanOne, beanTwo, "Beans with different load timeouts should not be equal");
		}

		[Test]
		[Category("EqualityPositiveTestGroup")]
		[Category("BeanEqualityPositiveTestGroup")]
		[Category("LoadableBeanEqualityPositiveTestGroup")]
		public void ShouldBeEqual() {
			ILoadableBean beanOne = new LoadableBean();
			ILoadableBean beanTwo = new LoadableBean();

			Assert.AreEqual(beanOne, beanTwo, "Two newly constructed beans should be equal before any setters are " + 
				"invoked");
			
			Mock<IWebDriver> mockDriver = new Mock<IWebDriver>();

			beanOne.Driver = mockDriver.Object;
			beanTwo.Driver = mockDriver.Object;

			Assert.AreEqual(beanOne, beanTwo, "Two beans should be equal if they have the same web driver reference " +
				"and the default load timeout");

			beanOne.LoadTimeout = 40;
			beanTwo.LoadTimeout = 40;

			Assert.AreEqual(beanOne, beanTwo, "Two beans should be equal if they have the same web driver reference " +
				"and the same non-default load timeout");

			beanOne = new LoadableBean();
			beanTwo = new LoadableBean();

			beanOne.LoadTimeout = 40;
			beanTwo.LoadTimeout = 40;

			Assert.AreEqual(beanOne, beanTwo, "Two beans should be equal if they have null web driver reference " +
				"and the same non-default load timeout");
		}
	}
}

